File: app/src/main/java/br/com/jonathanzanella/myexpenses/database/ModelRepository.java
package br.com.jonathanzanella.myexpenses.database;

import br.com.jonathanzanella.myexpenses.sync.UnsyncModel;
import br.com.jonathanzanella.myexpenses.validations.ValidationResult;

public interface ModelRepository<T extends UnsyncModel> {
	ValidationResult syncAndSave(final T unsyncAccount);
}

File: app/src/main/java/br/com/jonathanzanella/myexpenses/database/ModelRepository.kt
package br.com.jonathanzanella.myexpenses.database

import br.com.jonathanzanella.myexpenses.sync.UnsyncModel
import br.com.jonathanzanella.myexpenses.validations.ValidationResult

interface ModelRepository<in T : UnsyncModel> {
    fun syncAndSave(unsyncAccount: T): ValidationResult
}